app/services/update_shadow_team_association_service.rb

```diff
@@ -0,0 +1,37 @@
+class UpdateShadowTeamAssociationService
+  def self.update(team_code:, team_name:)
+    new(team_code: team_code, team_name: team_name).update
+  end
+
+  def initialize(team_code:, team_name:)
+    @team_code = team_code
+    @team_name = team_name
+    @team = nil
+  end
+
+  def update
+    retrieve_team if @team_name.match('(?i)OMIC.').present?
+
+    return Rails.logger.error("Couldn't find team with name #{@team_name}") if @team.nil?
+
+    update_shadow_code if incorrect_team_shadow_code
+  end
+
+private
+
+  def retrieve_team
```

Comment:
This can probably be allot cleaner

Reply:
@maysakanoni This does what we need it to do – so that's the first hurdle.

I think a regular expression could help reduce the number of lines and code repetition. This is an example where we'd want to extract data from the original string using a regular expression (using the `match` method) rather than just test whether it matches (with `.match?`).

I'm not sure how familiar you are with regular expressions, but thought I'd share this in case it's useful. I like using a tool called [RegExr](https://regexr.com/) when writing regular expressions – it has built in documentation, and an 'explain' feature which will describe exactly what the regular expression is doing and why.

I've put together an example of a regex that could be useful to when trying to extract an 'active' team name from the 'shadow' team. Have a look if you're interested: https://regexr.com/54kbn

In RegExr, the blue lines are the ones which have matched against the regular expression. If you hover over each highlighted line, you'll see a `Group #1` value – this is what we'd want to _extract_ from the original string using Ruby's `match` method.

Also note that this is a case-insensitive match, so will also be able to extract an active team name from a really weird shadow team (for example: `oMiC - Something` or `OmIc - Something`)